### PR TITLE
Clean up SQL Server transport "design" doco

### PR DESCRIPTION
**Pull request by SzymonPobiega**

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@weralabaj please review.

File: nservicebus/sqlserver/concurrency_content_sqlserver_[,2).partial.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Each endpoint running SQL Server transport spins up a fixed number of threads (controlled by `MaximumConcurrencyLevel` property of `TransportConfig` section) both for each input queue. Each thread runs in a loop, polling the database for messages awaiting processing.
```
Comment (weralabaj) [inline nservicebus/sqlserver/concurrency_content_sqlserver_[,2).partial.md]: > both for each input queue

~~both ~~

File: nservicebus/sqlserver/concurrency_content_sqlserver_[,2).partial.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+
+As a consequence, if `MaximumConcurrencyLevel` is set to 10, there are 40 threads running and constantly polling the database (10 threads for the main queue, 10 for each of satellite).
```
Comment (weralabaj) [inline nservicebus/sqlserver/concurrency_content_sqlserver_[,2).partial.md]: Satellite comes up out of the blue, with no introduction and no connection with the previous paragraph. 

How about:

By default, there are 4 input queues created for every endpoint: (list them, e.g. Timeouts queue,...). As a consequence, ....

> 10 for each of satellite

10 for each input queue

File: nservicebus/sqlserver/concurrency_content_sqlserver_[2,2.1).partial.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Each endpoint running SQL Server transport spins up a fixed number of threads (controlled by `MaximumConcurrencyLevel` property of `TransportConfig` section) both for each input queue. Each thread runs in a loop, polling the database for messages awaiting processing.
+
```
Comment (weralabaj) [inline nservicebus/sqlserver/concurrency_content_sqlserver_[2,2.1).partial.md]: ~~both~~

File: nservicebus/sqlserver/concurrency_content_sqlserver_[2,2.1).partial.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+As a consequence, if `MaximumConcurrencyLevel` is set to 10, there are 31 threads running and constantly polling the database (10 threads for the main queue, 10 for each satellite and another one for the callback receiver). The maximum concurrency value for the callback receiver can be adjusted via the configuration API.
```
Comment (weralabaj) [inline nservicebus/sqlserver/concurrency_content_sqlserver_[2,2.1).partial.md]: Seems in that case it's not exactly "for each input queue", callbacks is special ;)

File: nservicebus/sqlserver/concurrency_content_sqlserver_[2.1,3).partial.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+SQL Server transport uses an adaptive concurrency model. The transport adapts the number of polling threads based on the rate of messages coming in. The key concept in this new model is the *ramp up controller* which controls the ramping up of new threads and decommissioning of unnecessary threads. It uses the following algorithm:
+
+ * if last receive operation yielded a message, it increments the *consecutive successes* counter and resets the *consecutive failures* counter
```
Comment (weralabaj) [inline nservicebus/sqlserver/concurrency_content_sqlserver_[2.1,3).partial.md]: Also maybe we could just say something more generic, like that the algorithm adjusts the number of threads to the number of messages received in the given period of time (e.g. the higher the avg number of msmgs the higher the number of threads), without giving away all the details ;)

And say that it'll be never higher than `MaximumConcurrencyLevel`.

File: nservicebus/sqlserver/concurrency_content_sqlserver_[3,).partial.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+SQL Server transport maintains a dedicated monitoring thread for each input queue. It is responsible for detecting the number of messages waiting for delivery and creating receive [Task](https://msdn.microsoft.com/en-us/library/system.threading.tasks.task.aspx)s - one for each pending message.
+
+The maximum number of concurrent tasks will never exceed `MaximumConcurrencyLevel`. The number of tasks does not translate to the number of running threads which is controlled by the TPL scheduling mechanisms.
```
Comment (weralabaj) [inline nservicebus/sqlserver/concurrency_content_sqlserver_[3,).partial.md]: Worth a link to MSDN?

**Comment by SzymonPobiega** [comment]
@weralabaj I updated the content to reflect your comments.

File: nservicebus/sqlserver/transactions_native_sqlserver_[,3).partial.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
+There are two available options within native transaction level:
```
Comment (weralabaj) [inline nservicebus/sqlserver/transactions_native_sqlserver_[,3).partial.md]: seems those two partials should go the other way round? this one is the [3,x) right?

Comment (SzymonPobiega) [inline nservicebus/sqlserver/transactions_native_sqlserver_[,3).partial.md]: Yeah, I'll swap them

File: nservicebus/sqlserver/transactions_native_sqlserver_[3,).partial.md [hunk 8]
```diff
@@ -0,0 +1 @@
+There was no distinction between `ReceiveOnly` and `SendsAtomicWithReceive`. Using native transaction was equivalent to `SendsAtomicWithReceive` mode.
```
Comment (weralabaj) [inline nservicebus/sqlserver/transactions_native_sqlserver_[3,).partial.md]: I'd say :

"Using native transaction is equivalent to [`SendsAtomicWithReceive` mode in Version 3 and higher](/nservicebus/sqlserver/transactions.md?version=sqlserver_3#native-transactions).

Note the link above might be wrong, I made it up in my head :)

Comment (SzymonPobiega) [inline nservicebus/sqlserver/transactions_native_sqlserver_[3,).partial.md]: Let's try this link and correct if wrong :)

**Pull request merged**